content/articles/functional-programming-with-java-lambda/index.md

```diff
@@ -0,0 +1,333 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /functional-programming-with-java-lambda/
+title: Functional Programming with Java Lambda
+description: In this article, we will look at the concept of Functional programming in Java. we will also explain how a function can be used as a value.
+author: popoola-tunde
+date: 2021-10-12T00:00:00-10:03
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/functional-programming-with-java-lambda/hero.jpg
+    alt: Functional Programming with Java Lambda Hero Image
+---
+Java is an object-oriented programming language. It provides for the encapsulation of all variables and methods of the class. 
+<!--more-->
+That is, code blocks before Java version 8 are done with classes and objects. The process of passing a behavior to methods takes extra work and load. This is because we are passing a class with another attribute.
+
+In Java 8, Lambda was introduced to represent the functional interface of Java. This means that lambdas expression is the expression of functional interface. 
+
+Lambda was introduced to implement behavior code. This code makes it easier through the creation of a function that performs a specific task. Also, this code does not belong to any existing class thus treated as a value.
+
+In this article, the reader will be able to understand Java's functional programming and learn how to assign lambdas to an interface. We will work with lambda expressions using a different number of parameters. 
+
+The reader will be able to convert the object or a class's method to lambdas function with less code. The reader will also understand the importance of using functional programming.
+
+### Prerequisites
+To follow along with this tutorial, the reader should have:
+- A basic knowledge of Java.
+- [Java](https://www.oracle.com/java/technologies/downloads/)  installed.
+- Any Java IDE of their choice.
+- An understanding of class's and static methods.
+
+### Function as a value
```

Comment:
Use a descriptive heading, something that explains well what is happening in the heading.